### ISSUE TITLE
Trend of evaluation metrics and losses on TensorBoard

### ISSUE THREAD

**Issue by centiLinda** [StanfordASL/Trajectron-plus-plus#83]
Hi Boris,

When I visualize the evaluation metrics and losses recorded on TensorBoard, the trend between NLL, 50-samples, and Most-Likely are very different as shown below:
![metrics](https://user-images.githubusercontent.com/42902063/235602264-e60d4561-9265-452d-854e-7cadb07d76e1.jpg)
The training loss is converging normally:
![loss](https://user-images.githubusercontent.com/42902063/235602480-dcabf058-ce66-40a9-af8f-a4a0f931c5b1.jpg)

NLL and ADE/FDE with 50 samples are generally improving, but shows slightly better results during earlier epochs (around 60). Most-likely ADE/FDE are not showing meaningful trends (see the last 4 plots). Is this also the case in your experiments?

It's mentioned in the paper that you trained the model for 100 epochs for ETH dataset. Are the reported values all based on epoch-100? Or did you use the epoch with the best performance during training? If so, how did you select the best epoch (since the evaluation metrics do not show a consistent trend)?

**Comment by BorisIvanovic** [comment]
Hi @centiLinda, thanks for the question! We didn't do anything fancy, I believe we just used the model after 100 epochs. I didn't want to spend too much time optimizing things by selecting the best epoch and etc. (maybe to our detriment, but that's ok 🤷‍♂️). Feel very free to take the model at epoch 60 if you think it is better!

As for the metrics, I'm not too worried about the last 4 plots, since their y-axes ranges are quite small (you're seeing very small changes in the actual value being plotted, just that the y-axis limits are very close together).

Also, going with the above comment, it really is very possible that the hyperparameters we chose are not the best ones. For example, maybe larger batch sizes would be better, maybe different learning rates would be better, etc. 

I hope this helps!
